plugin/src/main/java/com/denizenscript/denizen/scripts/commands/entity/InvisibleCommand.java

```diff
@@ -1,60 +1,79 @@
 import com.denizenscript.denizencore.scripts.ScriptEntry;
 import com.denizenscript.denizencore.scripts.commands.AbstractCommand;
-import net.citizensnpcs.api.npc.NPC;
-import org.bukkit.entity.ArmorStand;
-import org.bukkit.entity.ItemFrame;
-import org.bukkit.potion.PotionEffect;
+import org.bukkit.entity.*;
 import org.bukkit.potion.PotionEffectType;
 
+import java.util.HashMap;
+import java.util.List;
+import java.util.UUID;
+
 public class InvisibleCommand extends AbstractCommand {
 
     public InvisibleCommand() {
         setName("invisible");
-        setSyntax("invisible [<entity>] (state:true/false/toggle)");
-        setRequiredArguments(1, 2);
+        setSyntax("invisible [<entity>] (state:true/false/toggle/reset) (for:<player>|...)");
+        setPrefixesHandled("for");
+        setRequiredArguments(0, 3);
         isProcedural = false;
     }
 
     // <--[command]
     // @Name Invisible
-    // @Syntax invisible [<entity>] (state:true/false/toggle)
-    // @Required 1
-    // @Maximum 2
-    // @Short Makes an NPC or entity go invisible.
+    // @Syntax invisible [<entity>] (state:true/false/toggle/reset) (for:<player>|...)
+    // @Required 0
+    // @Maximum 3
+    // @Short Sets whether an NPC or entity are invisible.
     // @Group entity
     //
     // @Description
-    // For living entities (other than armor_stand), applies a maximum duration invisibility potion.
-    // For armor_stands or item_frame, toggles them invisible.
-    // Applies the 'invisible' trait to NPCs.
+    // Makes the specified entity invisible, defaults to the linked player/NPC if one wasn't specified.
+    // If an NPC was specified, the 'invisible' trait is applied.
+    //
+    // Optionally specify 'for:' with a list of players to fake the entity's visibility state for these players.
+    // When using the 'toggle' state with the 'for:' argument, the visibility state will be toggled for each player separately.
+    // Note that using the 'for:' argument won't apply the 'invisible' trait to NPCs.
+    // If unspecified, will be set globally.
     //
-    // NPCs can't be made invisible if not added to the playerlist.
-    // (The invisible trait adds the NPC to the playerlist when set)
-    // See <@link language invisible trait>)
+    // To reset an entity's fake visibility (for all players) use the 'reset' state.
+    //
+    // NPCs can't be made invisible if not added to the playerlist (the invisible trait adds the NPC to the playerlist when set).
+    // See <@link language invisible trait>
     //
     // @Tags
     // None
     //
     // @Usage
-    // Use to makes the player invisible.
-    // - invisible <player> state:true
+    // Use to make the linked player (or NPC, if there isn't one) invisible.
+    // - invisible
     //
     // @Usage
-    // Use to make the attached NPC visible if previously invisible, and invisible if not
+    // Use to make the linked NPC visible if previously invisible, and invisible if not

```

Comment:
should add the missing period at end of line

Reply:
Added